test/simulation.jl

```diff
@@ -133,6 +133,22 @@ fl_res = copy(du.C)
 
 @test norm(fc_res - fp_res) < 1e-4
 @test norm(fc_res - fl_res) < 1e-4
+
+# Test same but with no preallocating
+
+f = evalsim(DiffusionWithLiteral, can_prealloc=false)
+f_noalloc = f(torus, generate)
+
+f = evalsim(DiffusionWithLiteral)
+f_alloc = f(torus, generate)
+
+f_noalloc(du, u₀, NamedTuple(), 0)
```

Comment:
Test `@allocations` as well

Reply:
Number of allocations for new is reasonable but I can't really understand the size of the allocations. It's less than I expect.